Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 50bcd4574006198480ed38745974b2c8a6699d9e
Build: success
Build: success
eclwatch/ClusterProcessesQueryWidget.js: line 321, col 30, PreflightDetailsWidget is not defined.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! eclwatch@1.0.0 lint: `jshint --config ./.jshintrc ./eclwatch`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the eclwatch@1.0.0 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?
npm ERR! A complete log of this run can be found in:
npm ERR!     /mnt/disk1/home/vamosax/.npm/_logs/2019-06-18T02_17_51_442Z-debug.log
ERROR: lint exited with 2.
npm ERR! Test failed.  See above for more details.
npm test end
~/smoketest/PR-12625/build
Makefiles created (2019-06-17_22-17-51 43 sec )
Build it
cmd: make -j   (2019-06-17_22-17-51)
Get ECLWatch build result
Build and Install finished (2019-06-17_22-21-21 210 sec)
Check the build result
Get ECLWatch build result

Lint error(s): 
eclwatch/ClusterProcessesQueryWidget.js: line 232, col 18, let is available in ES6 (use esversion: 6) or Mozilla JS extensions (use moz).

Install HPCC Platform
Number of Error(s): 4
2 errors
ModuleNotFoundError: Module not found: Error: Cant resolve hpcc/MachineInformationWidget in /mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/tmp/eclwatch
CMake Error at esp/src/cmake_install.cmake:88 (message):
make: *** [install] Error 1
CMake Error at esp/src/cmake_install.cmake:88 (message):
  npm run bundle failed see above and
  /mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/build_bundle.txt
  for details.
Call Stack (most recent call first):
  esp/cmake_install.cmake:46 (include)
  cmake_install.cmake:55 (include)
make: *** [install] Error 1
Number of Error(s): 4
2 errors
ModuleNotFoundError: Module not found: Error: Cant resolve hpcc/MachineInformationWidget in /mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/tmp/eclwatch
CMake Error at esp/src/cmake_install.cmake:88 (message):
make: *** [install] Error 1
CMakeFiles/CMakeOutput.log copied into ../CMakeOutput-2019-06-17_22-17-08.log
CMakeFiles/CMakeError.log copied into ../CMakeError-2019-06-17_22-17-08.log
Get ECLWatch build result
ECL Watch:  Rebuilding Site
---- Build Target: /mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/build
---- Creating out of source build environment
---- Installing third party libraries
---- Copy resources
---- Build website
ModuleNotFoundError: Module not found: Error: Cant resolve hpcc/MachineInformationWidget in /mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/tmp/eclwatch
    at factory.create (/mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/tmp/node_modules/webpack/lib/Compilation.js:823:10)
    at factory (/mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/tmp/node_modules/webpack/lib/NormalModuleFactory.js:397:22)
    at resolver (/mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/tmp/node_modules/webpack/lib/NormalModuleFactory.js:130:21)
    at resolver (/mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/tmp/node_modules/dojo-webpack-plugin/lib/DojoAMDModuleFactoryPlugin.js:328:5)
    at asyncLib.parallel (/mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/tmp/node_modules/webpack/lib/NormalModuleFactory.js:224:22)
    at /mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/tmp/node_modules/neo-async/async.js:2830:7
    at /mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/tmp/node_modules/neo-async/async.js:6877:13
    at normalResolver.resolve (/mnt/disk1/home/vamosax/smoketest/PR-12625/build/esp/src/tmp/node_modules/webpack/lib/NormalModuleFactory.js:214:25)
    at doResolv
 ... ( comment is too long, 169 lines, 13938 bytes, truncated) ... 
[Build error:](https://gist.github.com/9a1e12c31c0565ecfb77723be188b136)
Build log [Build log](https://gist.github.com/HPCCSmoketest/9a1e12c31c0565ecfb77723be188b136#file-result-19-06-17-22-16-43-log)

```

--- a/esp/src/eclwatch/ClusterProcessesQueryWidget.js
+++ b/esp/src/eclwatch/ClusterProcessesQueryWidget.js
@@ -1,0 +1,333 @@
+define([
+    "dojo/_base/declare",
+    "dojo/_base/lang",
+    "dojo/i18n",
+    "dojo/i18n!./nls/hpcc",
+    "dojo/_base/array",
+    "dojo/on",
+    "dojo/dom",
+    "dojo/dom-class",
+    "dojo/dom-construct",
+    "dojo/topic",
+
+    "dijit/registry",
+    "dijit/form/Button",
+    "dijit/ToolbarSeparator",
+    "dijit/Dialog",
+    "dijit/form/TextBox",
+
+    "dgrid/tree",
+    "dgrid/selector",
+
+    "hpcc/GridDetailsWidget",
+    "src/ESPPreflight",
+    "src/ESPRequest",
+    "src/WsTopology",
+    "src/Utility",
+    "src/ESPUtil",
+    "hpcc/DelayLoadWidget",
+    "hpcc/PreflightDetailsWidget",
+    "hpcc/MachineInformationWidget",
+    "hpcc/IFrameWidget"
+], function (declare, lang, i18n, nlsHPCC, arrayUtil, on, dom, domClass, domConstruct, topic,
+    registry, Button, ToolbarSeparator, Dialog, TextBox,
+    tree, selector,
+    GridDetailsWidget, ESPPreflight, ESPRequest, WsTopology, Utility, ESPUtil, DelayLoadWidget, PreflightDetailsWidget, MachineInformationWidget, IFrameWidget) {
+    return declare("CluserProcessesQueryWidget", [GridDetailsWidget, ESPUtil.FormHelper], {
+        i18n: nlsHPCC,
+
+        gridTitle: nlsHPCC.title_Clusters,
+        idProperty: "__hpcc_id",
+        machineFilter: null,
+        machineFilterLoaded: null,
+
+        postCreate: function (args) {
+            var context = this;
+            this.inherited(arguments);
+            this.openButton = registry.byId(this.id + "Open");
+            this.refreshButton = registry.byId(this.id + "Refresh");
+            this.configurationButton = registry.byId(this.id + "Configuration");
+
+            this.machineFilter = new MachineInformationWidget({});
+
+            this.configurationButton = new Button({
+                label:  this.i18n.OpenConfiguration,
+                onClick: function(event) {
+                    context._onOpenConfiguration()
+                }
+            });
+
+            this.machineFilter.placeAt(this.openButton.domNode, "after");
+            this.configurationButton.placeAt(this.openButton.domNode, "after");
+
+            new ToolbarSeparator().placeAt(this.machineFilter.domNode, "before");
+
+            this.machineFilter.machineForm.set("style", "width:500px;");
+            this.machineFilter.disable(true);
+
+            this.legacyClustersProcessesIframeWidget = new IFrameWidget({
+                id: this.id + "_LegacyClustersProcessesIframeWidget",
+                title: "Clusters Processes (legacy)",
+                style: "border: 0; width: 100%; height: 100%"
+            });
+            this.legacyClustersProcessesIframeWidget.placeAt(this._tabContainer, "last");
+        },
+
+        init: function (params) {
+            var context = this;
+            if (this.inherited(arguments))
+                return;
+
+            this.refreshActionState();
+            this.refreshGrid();
+
+            this.machineFilter.on("apply", function (evt) {
+                var selection = context.grid.getSelected();
+                var selections = [];
+                for (var i = 0; i < selection.length; ++i) {
+                    var data = context.grid.row(selection[i].hpcc_id).data;
+                    selections.push(data);
+                }
+                context.machineFilter._onSubmitRequest("machine",  selections);
+            });
+
+            topic.subscribe("createClusterProcessPreflightTab", function (topic) {
+                var pfTab = context.ensureMIPane(topic.response.Machines.MachineInfoEx[0].DisplayType + "- " + topic.response.TimeStamp, {
+                    params: topic.response
+                });
+                pfTab.init(topic.response, "machines");
+            });
+
+            dojo.destroy(this.id + "Open");
+        },
+
+        initTab: function () {
+            var currSel = this.getSelectedChild();
+            if (currSel && !currSel.initalized) {
+                if (currSel.id === this.id + "_Grid") {
+                    this.refreshGrid()
+                } else if (currSel.id === this.legacyClustersProcessesIframeWidget.id && !this.legacyClustersProcessesIframeWidget.initalized) {
+                    this.legacyClustersProcessesIframeWidget.init({
+                        src: ESPRequest.getBaseURL("WsTopology") + "/TpClusterQuery?Type=ROOT"
+                    });
+                } else {
+                    currSel.init(currSel.params);
+                }
+            }
+        },
+
+        _onRowDblClick: function (item) {
+            var nodeTab = this.ensureLogsPane(item.Name, {
+                params: item,
+                ParentName: item.Parent.Name,
+                LogDirectory: item.Parent.LogDirectory,
+                NetAddress: item.Netaddress,
+                OS: item.OS,
+                newPreflight: true
+            });
+            this.selectChild(nodeTab);
+        },
+
+        createGrid: function (domID) {
+            var context = this;
+            var retVal = new declare([ESPUtil.Grid(true, true, false, true)])({
+                store: ESPPreflight.CreateClusterProcessStore(),
+                columns: {
+                    col1: selector({
+                        width: 20,
+                        selectorType: 'checkbox'
+                    }),
+                    Configuration: {
+                        label: this.i18n.Configuration,
+                        renderHeaderCell: function (node) {
+                            node.innerHTML = Utility.getImageHTML("configuration.png", context.i18n.Configuration);
+                        },
+                        width: 10,
+                        sortable: false,
+                        formatter: function (clusterType) {
+                            if (clusterType === true) {
+                                return "<a href='#' />" + Utility.getImageHTML("configuration.png", context.i18n.Configuration) + "</a>";
+                            }
+                            return "";
+                        }
+                    },
+                    Name: tree({
+                        formatter: function (_name, row) {
+                            var img = "";
+                            var name = _name;
+                            if (row.type === "clusterProcess") {
+                                img = Utility.getImageHTML("server.png");
+                                name = row.Type + " - " + _name;
+                            } if (row.type === "machine") {
+                                img = Utility.getImageHTML("machine.png");
+                                name = "<a href='#' class='dgrid-row-url'>" + row.Netaddress + " - " + _name + "</a>";
+                            }
+                            return img + "&nbsp;" + name;
+                        },
+                        collapseOnRefresh: false,
+                        label: this.i18n.Name,
+                        sortable: true,
+                        width: 150
+                    }),
+                    Domain: {
+                        label: this.i18n.Domain,
+                        sortable: false,
+                        width: 100
+                    },
+                    Platform: {
+                        label: this.i18n.Platform,
+                        sortable: false,
+                        width: 75
+                    },
+                    ProcessNumber: {
+                        label: this.i18n.SlaveNumber,
+                        sortable: false,
+                        width: 100
+                    },
+                    Directory: {
+                        label: this.i18n.Directory,
+                        sortable: false,
+                        width: 200
+                    },
+                    LogDirectory: {
+                        label: this.i18n.LogDirectory,
+                        sortable: false,
+                        width: 200
+                    }
+                }
+            }, domID);
+
+            retVal.on(".dgrid-row-url:click", function (evt) {
+                if (context._onRowDblClick) {
+                    var item = retVal.row(evt).data;
+                    context._onRowDblClick(item);
+                }
+            });
+
+            retVal.on(".dgrid-row:dblclick", function (evt) {
+                if (context._onRowDblClick) {
+                    var item = retVal.row(evt).data;
+                    context._onRowDblClick(item);
+                }
+            });
+
+            retVal.on(".dgrid-cell:click", function(evt){
+                var cell = retVal.cell(evt)
+            });
+
+            retVal.on("dgrid-select", function (event) {
+                context.refreshActionState();
+            });
+            retVal.on("dgrid-deselect", function (event) {
+                context.refreshActionState();
+            });
+
+            return retVal;
+        },
+
+        _onOpenConfiguration: function (event) {
+            var context = this;
+            var selections = this.grid.getSelected();
+            var firstTab = null;
+            for (var i = 0; i < selections.length; ++i) {
+                WsTopology.TpGetComponentFile({
+                    request: {
+                        FileType: "cfg",
+                        CompType: selections[i].Component,
+                        CompName: selections[i].Name,
+                        Directory: selections[i].Directory,
+                        OsType: selections[i].OS
+                    }
+                }).then(function(response){
+                    var tab = context.ensureConfigurationPane(selections[i].Component + "_" + selections[i].Name , {
+                        Component: selections[i].Component,
+                        Name: selections[i].Name,
+                        Usergenerated: response
+                    });
+                });
+            }
+        },
+
+        _onRefresh: function () {
+            this.refreshGrid();
+        },
+
+        refreshGrid: function () {
+            this.grid.set("query", {
+                Type: "ROOT"
+            });
+        },
+
+        refreshActionState: function () {
+            var selection = this.grid.getSelected();
+            var isTarget = false;
+            var isProcess = false;
+
+            for (var i = 0; i < selection.length; ++i) {
+                if (selection[i] && selection[i].type === "clusterProcess") {
+                    isTarget = true;
+                    isProcess = false;
+
+                } else {
+                    isTarget = false;
+                    isProcess = true;
+                }
+            }
+
+            this.machineFilter.disable(!isProcess);
+            this.configurationButton.set("disabled", !isTarget);
+        },
+
+        ensureConfigurationPane: function (id, params) {
+            id = this.createChildTabID(id);
+            var retVal = registry.byId(id);
+            if (!retVal) {
+                var context = this;
+                retVal = new DelayLoadWidget({
+                    id: id,
+                    title: "<b>" + params.Component + "</b>: " + params.Name + " " + context.i18n.Configuration,
+                    closable: true,
+                    delayWidget: "ECLSourceWidget",
+                    params: params
+
+                });
+                this.addChild(retVal, "last");
+            }
+            return retVal;
+        },
+
+        ensureLogsPane: function (id, params) {
+            id = this.createChildTabID(id);
+            var retVal = registry.byId(id);
+            if (!retVal) {
+                var context = this;
+                retVal = new DelayLoadWidget({
+                    id: id,
+                    title: "<b>" + params.ParentName + "</b>: " + params.NetAddress,
+                    closable: true,
+                    delayWidget: "LogWidget",
+                    params: params
+
+                });
+                this.addChild(retVal, "last");
+            }
+            return retVal;
+        },
+
+        ensureMIPane: function (id, params) {
+            id = this.createChildTabID(id);
+            var retVal = registry.byId(id);
+            if (!retVal) {
+                retVal = new PreflightDetailsWidget({
+                    id: id,
+                    style: "width: 100%",
+                    params: params.params,
+                    closable: true,
+                    title: params.params.Machines.MachineInfoEx[0].DisplayType
+                });
+                this._tabContainer.addChild(retVal, "last");
+            }
+            return retVal;
+        },
+    });
+});

--- a/esp/src/eclwatch/HPCCPlatformOpsWidget.js
+++ b/esp/src/eclwatch/HPCCPlatformOpsWidget.js
@@ -62,12 +62,7 @@ define([
             initTab: function () {
                 var currSel = this.getSelectedChild();
                 if (currSel && !currSel.initalized) {
-                    if (currSel.id === this.id + "_ClusterProcesses") {
-                        currSel.set("content", dojo.create("iframe", {
-                            src: dojoConfig.urlInfo.pathname + "?Widget=IFrameWidget&src=" + encodeURIComponent(ESPRequest.getBaseURL("WsTopology") + "/TpClusterQuery?Type=ROOT"),
-                            style: "border: 0; width: 100%; height: 100%"
-                        }));
-                    } else if (currSel.id === this.id + "_SystemServers") {
+                    if (currSel.id === this.id + "_SystemServers") {
                         currSel.set("content", dojo.create("iframe", {
                             src: dojoConfig.urlInfo.pathname + "?Widget=IFrameWidget&src=" + encodeURIComponent(ESPRequest.getBaseURL("WsTopology") + "/TpServiceQuery?Type=ALLSERVICES"),
                             style: "border: 0; width: 100%; height: 100%"

--- a/esp/src/eclwatch/templates/HPCCPlatformOpsWidget.html
+++ b/esp/src/eclwatch/templates/HPCCPlatformOpsWidget.html
@@ -10,7 +10,7 @@
             </div>
             <div id="${id}_TargetClustersQuery" title="${i18n.TargetClusters}" style="padding: 0px; border:0px; border-color:none; overflow: hidden" data-dojo-props="delayWidget: 'TargetClustersQueryWidget'" data-dojo-type="DelayLoadWidget">
             </div>
-            <div id="${id}_ClusterProcesses" title="${i18n.ClusterProcesses}" style="padding: 0px; border:0px; border-color:none; overflow: hidden" data-dojo-type="dijit.layout.ContentPane">
+            <div id="${id}_ClusterProcessesQuery" title="${i18n.ClusterProcesses}" style="padding: 0px; border:0px; border-color:none; overflow: hidden" data-dojo-props="delayWidget: 'ClusterProcessesQueryWidget'" data-dojo-type="DelayLoadWidget">
             </div>
             <div id="${id}_SystemServers" title="${i18n.SystemServers}" style="padding: 0px; border:0px; border-color:none; overflow: hidden" data-dojo-type="dijit.layout.ContentPane">
             </div>

--- a/esp/src/src/Utility.ts
+++ b/esp/src/src/Utility.ts
@@ -400,6 +400,9 @@ export function resolve(hpccWidget, callback) {
         case "ActivityWidget":
             require(["hpcc/ActivityWidget"], doLoad);
             break;
+        case "ClusterProcessesQueryWidget":
+            require(["hpcc/ClusterProcessesQueryWidget"], doLoad);
+            break;
         case "CurrentUserDetailsWidget":
             require(["hpcc/CurrentUserDetailsWidget"], doLoad);
             break;

